app/models/dor_object_workflow_status.rb

```diff
@@ -0,0 +1,24 @@
+class DorObjectWorkflowStatus
+  attr_reader :pid
+
+  ##
+  # @param [String] pid in format "druid:abc123def4567"
+  def initialize(pid)
+    @pid = pid
+  end
+
+  ##
+  # @return [Boolean]
+  def can_open_version?
+    return false unless workflow.get_lifecycle('dor', pid, 'accessioned')
+    return false if workflow.get_active_lifecycle('dor', pid, 'submitted')
+    return false if workflow.get_active_lifecycle('dor', pid, 'opened')
+    true
+  end
+
+  private
+
+  def workflow
+    Dor::Config.workflow.client
+  end
+end
```

Comment:
Yes that's definitely coming when I integrate this. This PR solves the problem at hand, but leaves things ready for the integration.